### DISCUSSION TITLE
What language version does this version of DnSpy support?

### DISCUSSION THREAD

**Discussion by KieranDevvs** [dnSpyEx/dnSpy#149] [Q&A] [answered]
As the title asks, thanks in advance.

**Comment by ElektroKill** [comment, marked as answer]
This depends on what component you mean. If you mean the decompiler then it does not support many of the new C# language features as dnSpyEx uses a modified version of the ILSpy 2.4 decompiler engine. The language feature support in the decompiler was improved slightly over dnSpy but don't expect anything drastic. As for the features which allow the user to enter code, for example, C# Interactive, Edit Method (C#), debugger watch window, etc. they support the latest C# version supported by Roslyn 4.1.0 (for the 6.2.0 release) or Roslyn 4.4.0 for the latest builds from the `master` branch.

**Reply by KieranDevvs** [reply]
I was asking about decompilation really. I had issues with the old DnSpy where id go to edit a method and then compile it but it fails because the initial decompilation didn't produce a compilable state. Is there any plans to move to a newer version of ILSpy? I'd be willing to take a look.

**Reply by ElektroKill** [reply]
Updating the ILSpy engine is being worked on. See https://github.com/dnSpyEx/dnSpy/issues/5 for more information and the difficulties associated with it. TLDR: dnSpy needs ILSpy to output much more verbose information then it does by default. This needs heavy modification and adaptation of ILSpy. I am also waiting for ILSpy 8 to release as it contains some improvements and fixes which are quite vital for dnSpy.

**Reply by KieranDevvs** [reply]
Ok thanks for the update, i'll check out that issue.
